documentation/universal_links.md

```diff
@@ -0,0 +1,250 @@
+  - [Setting Up an Entitlements File for Universal Links](#setting-up-an-entitlements-file-for-universal-links)
+    - [Steps to Configure the Entitlements File](#steps-to-configure-the-entitlements-file)
+    - [Points to Consider](#points-to-consider)
+  - [Configuring Universal Links for your Storefront](#configuring-universal-links-for-your-storefront)
+  - [Important: The catch-all route `*`](#important-the-catch-all-route-)
+  - [Handling Universal Links in your app](#handling-universal-links-in-your-app)
+  - [Security Considerations](#security-considerations)
+  - [Further Reading and Resources](#further-reading-and-resources)
+
+## What are Universal Links?
+
+Universal Links are an iOS feature which allow apps to open links directly in an app rather than sending the user to a browser. This feature provides a seamless user experience by directly engaging app users with relevant content.
+
+## Setting Up an Entitlements File for Universal Links
+
+To enable Universal Links in your iOS app, you'll need to configure an entitlements file that specifies the associated domains your app is allowed to handle. Here’s how you can set it up:
+
+### Steps to Configure the Entitlements File
+
+1. **Open Your Xcode Project**:
+
+   - Launch Xcode and open your project or workspace file.
+
+2. **Navigate to Your App Target's Capabilities**:
+
+   - Click on your project in the Project Navigator.
+   - Select your app target from the list of targets.
+   - Click on the "Capabilities" tab at the top of the editor.
+
+3. **Enable Associated Domains**:
+
+   - In the Capabilities section, find "Associated Domains" and toggle the switch to enable it.
+   - A new section should appear to configure your associated domains.
+
+4. **Add Your Associated Domains**:
+
+   - Add each domain you want to associate with your app in the format: `applinks:yourdomain.com`
+     - For example, `applinks:example.com` if your domain is `example.com`.
+
+5. **Verify the Entitlements File**:
+
+   - Xcode will automatically add an entitlements file to your project if it doesn't exist, or update your existing one. You can verify this by checking for a file named `YourApp.entitlements` in your project.
+   - Open this file to ensure the associated domains are listed correctly under the `<key>com.apple.developer.associated-domains</key>` entry.
+
+6. **Build and Run Your App**:
+   - Make sure your app's provisioning profile includes support for the associated domains capability.
+   - Build and run your app on a device to test the setup.
+
+### Points to Consider
+
+- **HTTPS Requirement**: Ensure all of your associated domains are served over HTTPS.
+- **Multiple Domains**: You can list multiple associated domains by adding each one on a new line in the Associated Domains section within Xcode.
+- **Testing**: After setting up, test your Universal Links by opening valid URLs on a device to ensure they correctly redirect to your app.
```

Comment:
Might be useful to mention that theres two main ways to test here, either clicking a link in the notes app or something like that, or via CLI (e.g. `xcrun simctl openurl booted https://www.example.com/faq` or a cross platform solution for android too: https://github.com/expo/expo/tree/main/packages/uri-scheme#readme) 

and a call out to specify that pasting it in safari will fail?

Reply:
These settings are not available in the sim unfortunately